packages/react/src/modules/in-app-message/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,315 @@
+import {
+  ColorMode,
+  UseInAppMessageOptions,
+  useInAppChannel,
+  useInAppMessage,
+} from "@knocklabs/react-core";
+import * as Dialog from "@radix-ui/react-dialog";
+import clsx from "clsx";
+import React, { useEffect } from "react";
+
+import { ActionContent } from "../types";
+
+import "./styles.css";
+
+const MESSAGE_TYPE = "modal";
+
+export interface ModalProps {
+  filters?: UseInAppMessageOptions;
+}
+
+export interface ModalContent {
+  title: string;
+  body: string;
+  primary_button?: {
+    text: string;
+    action: string;
+  };
+  secondary_button?: {
+    text: string;
+    action: string;
+  };
+  dismissible?: boolean;
+}
+
+type RootProps = Omit<
+  React.ComponentPropsWithoutRef<typeof Dialog.Root>,
+  "modal"
+> &
+  React.PropsWithChildren<React.ComponentPropsWithRef<"div">>;
+
+const Root = ({ children, onOpenChange, ...props }: RootProps) => {
+  return (
+    <Dialog.Root defaultOpen onOpenChange={onOpenChange} {...props}>
+      <Dialog.Portal>{children}</Dialog.Portal>
+    </Dialog.Root>
+  );
+};
+
+type OverlayProps = React.ComponentPropsWithoutRef<typeof Dialog.Overlay> &
+  React.ComponentPropsWithRef<"div">;
+type OverlayRef = React.ElementRef<"div">;
+
+// TODO: Causes layout shift...
```

Comment:
Causes a bit of layout shift in the example app which I think is from the pointer / scroll  lock. Tested a few things out but haven't figured out exactly how to fix it yet.

Reply:
Have not run it locally to check out how/what it looks like, but as long as it's ticketed out or on your TODO to come back to, it doesn't seem like a blocker.